ecommerce_integrations/shopify/order.py

```diff
@@ -185,37 +186,46 @@ def _get_total_discount(line_item) -> float:
 	return sum(flt(discount.get("amount")) for discount in discount_allocations)
 
 
-def get_order_taxes(shopify_order, setting):
-	taxes = []
+def get_order_taxes(shopify_order, setting, items):
+	tax_account_wise_data = {}
 	line_items = shopify_order.get("line_items")
 
 	for line_item in line_items:
 		item_code = get_item_code(line_item)
 		for tax in line_item.get("tax_lines"):
-			taxes.append(
-				{
+			account_head = get_tax_account_head(tax)
+			tax_account_wise_data.setdefault(
+				account_head, {
 					"charge_type": "Actual",
-					"account_head": get_tax_account_head(tax),
 					"description": (
-						f"{get_tax_account_description(tax) or tax.get('title')} - {tax.get('rate') * 100.0:.2f}%"
+						f"{get_tax_account_description(tax) or tax.get('title')}"
 					),
-					"tax_amount": tax.get("price"),
-					"included_in_print_rate": 0,
 					"cost_center": setting.cost_center,
-					"item_wise_tax_detail": json.dumps(
-						{item_code: [flt(tax.get("rate")) * 100, flt(tax.get("price"))]}
-					),
+					"included_in_print_rate": 0,
 					"dont_recompute_tax": 1,
+					"tax_amount": 0,
+					"item_wise_tax_detail": {}
 				}
 			)
+			tax_account_wise_data[account_head]["tax_amount"] += flt(tax.get("price"))
+			tax_account_wise_data[account_head]["item_wise_tax_detail"].update({
+				item_code: [flt(tax.get("rate")) * 100, flt(tax.get("price"))]
+			})
 
-	taxes = update_taxes_with_shipping_lines(
-		taxes,
+	update_taxes_with_shipping_lines(
+		tax_account_wise_data,
 		shopify_order.get("shipping_lines"),
 		setting,
+		items,
 		taxes_inclusive=shopify_order.get("taxes_included"),
 	)
 
+	taxes = []
+	for account, tax_row in tax_account_wise_data.items():
+		row = {"account_head": account, **tax_row}
+		row["item_wise_tax_detail"] = json.dumps(row.get("item_wise_tax_detail", {}))
+		taxes.append(row)
+
 	return taxes
```

Comment:
Existing code is much more readable and we are adding a layer of complexity. It's better to break this and make it more readable.

Given a chance, I wouldn't touch existing functionality, but instead process the list of taxes just before returning it in a separate function (if applicable). Appending to items in `update_taxes_with_shipping_lines` would still be required though.

Reply:
Makes sense, have made consolidation an optional feature and also split it into a different function.